sqlobject/src/main/java/org/jdbi/v3/sqlobject/transaction/Transaction.java

```diff
@@ -62,10 +73,21 @@ public Handler decorateHandler(Handler base, Class<?> sqlObjectType, Method meth
 
                 HandleCallback<Object, Exception> callback = th -> base.invoke(target, args, handle);
 
-                if (isolation == TransactionIsolationLevel.INVALID_LEVEL) {
-                    return h.inTransaction(callback);
-                } else {
-                    return h.inTransaction(isolation, callback);
+                final boolean flipReadOnly = readOnly != h.isReadOnly();
+                if (flipReadOnly) {
+                    h.setReadOnly(readOnly);
+                }
```

Comment:
Can we add a check to the `if (h.isInTransaction())` block above? I think it's appropriate to throw an exception if the already-started transaction is read-write, but the present annotation's `readOnly()` is true.

We can probably ignore the inverse case.